src/main/java/dev/suse/explore/cap/CloudFoundryAPI.java

```diff
@@ -37,6 +44,35 @@ public boolean userAlreadyExists(String email) {
 	public String buildEnvironmentForUser(String email) throws Exception{
 		CloudFoundryOperations ops = context.getBean(CloudFoundryOperations.class);
 		
+		//**********create the org***************
+		//create org name from email: foo.bar@bar.com -> foor_bar_bar_com
+		String orgname = email.replace("\w", "_");
+
+		//How do I tell which parameters the request needs?
+		//TODO: replace quota name string with an env variable or something
+		CreateOrganizationRequest req = CreateOrganizationRequest.builder().organizationName(orgname).quotaDefinitionName("sandbox").build();
+		//Do I need to call .block() in the end? Why if so?
+		ops.organizations().create(req).block();
+
+		//**********make user org manager and developer*************
+		SetOrganizationRoleRequest req = SetOrganizationRoleRequest.builder().organizationName(orgname).organizationRole("Manager").build();
+		ops.useradmin().setOrganizationRole(req).block();
+		SetOrganizationRoleRequest req = SetOrganizationRoleRequest.builder().organizationName(orgname).organizationRole("Developer").build();
+		ops.useradmin().setOrganizationRole(req).block();
+
+		//***********create the default spaces************
+		//TODO: there is probably an elegant way to fold these into one call somehow
+		//TODO: Does the user get roles assigned for these spaces automatically?
+		CreateSpaceRequest req = CreateSpaceRequest.builder().name("dev").organization(orgname).build();
+		ops.spaces().create(req).block();
+		CreateSpaceRequest req = CreateSpaceRequest.builder().name("test").organization(orgname).build();
+		ops.spaces().create(req).block();
+		CreateSpaceRequest req = CreateSpaceRequest.builder().name("prod").organization(orgname).build();
+		ops.spaces().create(req).block();
+		CreateSpaceRequest req = CreateSpaceRequest.builder().name("samples").organization(orgname).build();
+		ops.spaces().create(req).block();
```

Comment:
I think we might want to time these calls and see if it's worth calling these and then blocking on each? It might also be worth spinning this whole class out into a Runnable so we can go ahead and pass the user back to WP before completion?

Reply:
Key question is if we want to wait until we know everything has worked or if we want to get back immediately and handle errors elsewhere. 

I'm currently reading up on some of this Reactive Programming paradigm stuff. There's a way to combine multiple Monos into one by using Mono<T>.zip(). The library uses this extensively for things that in my understanding are sort of atomic operations. Maybe that's the way